templates/rector-github-action-check.yaml

```diff
@@ -10,12 +10,20 @@ jobs:
         if: github.repository == '__CURRENT_REPOSITORY__'
         runs-on: ubuntu-latest
         steps:
+            # workaround for missing secret in fork PRs - see https://github.com/actions/checkout/issues/298
+            # see https://github.com/rectorphp/rector/commit/d395e1c28b8e6a56711dcc2e10490a82965850e4
             -
+                if: github.event.pull_request.head.repo.full_name == github.repository
                 uses: actions/checkout@v3
                 with:
-                    # needed to trigger workflow after push
+                    # Must be used to trigger workflow after push
                     token: ${{ secrets.ACCESS_TOKEN }}
 
+            # in forks, the token is not available - so we cannot use it
+            -
+                if: github.event.pull_request.head.repo.full_name != github.repository
+                uses: actions/checkout@v3
```

Comment:
THis should never run, as this conditoin is never met:

```bash
if: github.repository == '__CURRENT_REPOSITORY__'
```

Reply:
That seems overlapped, see https://github.com/rectorphp/rector-src/actions/runs/4305120316/jobs/7507111232#step:2:1

See issue on https://github.com/actions/checkout/issues/298

This is workaround until we get correct fix.

Also that needed on getrector-com repo, see skipped vs working

- https://github.com/rectorphp/getrector-com/actions/runs/4297578567/jobs/7490727969
- https://github.com/rectorphp/getrector-com/actions/runs/4308381219/jobs/7514599199

Reply:
@TomasVotruba looking at other repo implementation, the if seems before steps instead of before strategy

https://github.com/PostHog/posthog/blob/e5c44a044f316c41223da3ee6aad2234746a2522/.github/workflows/storybook-deploy.yml#L12

That's seems the reason why steps was always executed

Reply:
These miss-steps should be prevent by the `if`. How does the Google Acitons docs address this?

Maybe we have to define it on the more parent configuration?

Reply:
So what is the best way to disable whole workflow with one condition?

Reply:
I think the issue is that when GitHub runs a `pull_request` event it actually runs it in the base repository on a temporary branch which I think is a merge of the target and the head. Thus the `github.repository == 'rectorphp/rector-src'` returns true.

You need to check the repository or the head branch specifically - as has been done in the individual steps `github.event.pull_request.head.repo.full_name`. I'm not sure if there's a shorter variable to get it or not

Reply:
@jackbentley Thanks for the tip, I'll try that :+1: 

Is there a way to dump all the github metadata in workflow? It might be easier to just see the values there :)

Reply:
@jackbentley Seem it's not the right value, the Rector is skipped now:
https://github.com/rectorphp/rector-src/pull/3439

Reply:
I know if you enable debug it will output/let you download a dump of the entire `github` variable. I've seen you might be able to enable it on re-run via the action page also. I think it gives you the option of "re running with debug enabled" or something like that - haven't used that way myself though

https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging